### ISSUE TITLE
R CMD check signals a 7Mb R directory

### ISSUE THREAD

**Issue by dankelley** [ArgoCanada/argoFloats#95]
Indeed, when I see as follows.
```
ls -lh ~/Library/R/4.0/library/argoFloats/R
total 16408
-rw-r--r--  1 kelley  staff   1.0K 16 May 15:21 argoFloats
-rw-r--r--  1 kelley  staff   7.0M 16 May 15:21 argoFloats.rdb
-rw-r--r--  1 kelley  staff   2.0K 16 May 15:21 argoFloats.rdx
```
which is weird because the R code is only 100Kb.
```
du -k ~/git/argoFloats/R
100	/Users/kelley/git/argoFloats/R
```

This matters because `R CMD check` produces NOTEs for directories over 1M, and I could foresee problems in submitted the package to CRAN.

I may make a minimal package and see if this happens even there.  When I know more, and if I am still at a loss as to how to fix this problem, I will ask a question on R-devel.

**Comment by dankelley** [comment]
I'm starting to wonder if this is just a macos thing.  I did `devtools::check_win()` to do a check no windows, and in its output I see
```
* checking installed package size ... OK
```
which may mean that the size on windows is under 1M, since on macos that is the criterion for putting a NOTE after this line.

**Comment by dankelley** [comment]
Now it's up to 8M.

**Comment by dankelley** [comment]
@richardsc any chance you could do a linux build to see if it gives this >5Mb note?

I don't like to bug you but I didn't have any success when I tried this weekend.
* the work linux machines are running R 3.3, and I do not want to as the sysadmin to make a change during the lockdown.  (It can be a lot of work updating, because there are packages that might need updating at the sysadmin level.  And some package have broken, e.g. the BVP integrator I discussed in OAR did not even work in R 3.5.)
* I could not succeed in doing tests myself on mac with vbox.  Ubuntu refused to install, dying at the check-timezone step and burning up my CPUs for an hour or so, doing some stupid infinite loop or something. Debian -- I don't remember, but something went wrong.  Mint -- it actually installed, but the `apt` commands to update gfortran etc got into a circular loop of dependencies that I could not break through. This was with the most recent vbox, and I don't recommend installing it, since things were crazy laggy (10s response to typing a key or moving the mouse), I think because it was writing to a log file about every 10 microseconds.

**Comment by dankelley** [comment]
I tried an rhub build, with results as below.    Note the `checking installed package size` line, which is not followed by a message.  I do not really know what is going on with an rhub build, though ... maybe it has a threshold for reporting that's different from that on macos.

But, so far, what I have is
* macos: large-size warning that suggests impossibility of CRAN acceptance
* winbuilder: no such warning
* rhub/linux: no such warning

```
> library(rhub)
> check_on_linux()
─  Building package
─  Uploading package
─  Preparing build, see status at
   https://builder.r-hub.io/status/argoFloats_0.1.3.tar.gz-db26bd2bc5034127a04f55fca9dcb1d6
─  Build started
─  Downloading and unpacking package file
─  Querying system requirements
─  Installing system requirements
─  Starting Docker container
─  Querying package dependencies
─  Installing package dependencies
─  Running R CMD check
   About to run xvfb-run R CMD check argoFloats_0.1.3.tar.gz
─  using log directory ‘/home/docker/argoFloats.Rcheck’ (3.2s)
─  using R version 4.0.0 (2020-04-24)
─  using platform: x86_64-pc-linux-gnu (64-bit)
─  using session charset: UTF-8
✓  checking for file ‘argoFloats/DESCRIPTION’
─  checking extension type ... Package
─  this is package ‘argoFloats’ version ‘0.1.3’
─  package encoding: UTF-8
✓  checking package namespace information
✓  checking package dependencies (3.2s)
✓  checking if this is a source package
✓  checking if there is a namespace
✓  checking for executable files
✓  checking for hidden files and directories
✓  checking for portable file names
✓  checking for sufficient/correct file permissions
✓  checking whether package ‘argoFloats’ can be installed (1m 24.8s)
✓  checking installed package size
✓  checking package directory
✓  checking ‘build’ directory
✓  checking DESCRIPTION meta-information
✓  checking top-level files
✓  checking for left-over files
✓  checking index information (3.2s)
✓  checking package subdirectories
✓  checking R files for non-ASCII characters
✓  checking R files for syntax errors
✓  checking whether the package can be loaded (6.7s)
✓  checking whether the package can be loaded with stated dependencies (7.3s)
✓  checking whether the package can be unloaded cleanly (3.3s)
✓  checking whether the namespace can be loaded with stated dependencies (5.9s)
✓  checking whether the namespace can be unloaded cleanly (6.4s)
✓  checking loading without being on the library search path (3.2s)
✓  checking dependencies in R code (3.2s)
✓  checking S3 generic/method consistency (6.1s)
✓  checking replacement functions (6.5s)
✓  checking foreign function calls (5.7s)
✓  checking R code for possible problems (22.5s)
✓  checking Rd files
✓  checking Rd metadata
✓  checking Rd cross-references
✓  checking for missing documentation entries (6.7s)
✓  checking for code/documentation mismatches (17.6s)
✓  checking Rd \usage sections (8.8s)
✓  checking Rd contents
✓  checking for unstated dependencies in examples
✓  checking contents of ‘data’ directory
✓  checking data for non-ASCII characters (6.2s)
✓  checking data for ASCII and uncompressed saves
✓  checking installed files from ‘inst/doc’
✓  checking files in ‘vignettes’
✓  checking examples (15.3s)
✓  checking for unstated dependencies in ‘tests’
─  checking tests
✓  Running ‘testthat.R’ (9.4s)
✓  checking for unstated dependencies in vignettes (10.5s)
✓  checking package vignettes in ‘inst/doc’
─  checking running R code from vignettes
     ‘argoFloats.Rmd’ using ‘UTF-8’... OK
    NONE
✓  checking re-building of vignette outputs (37.7s)
✓  checking PDF version of manual (6.7s)
   
─  Done with R CMD check
─  Saving artifacts
```

**Comment by dankelley** [comment]
I've made some progress on this, in the "shrink-package" branch.  By "some" I mean a factor of 23.6, i.e. the package's "R" directory (which holds a database of R functions, and which is the thing that is tested during the build-check procedure) is now 323951 bytes in size, compared to the 7630639 that it was before.  It's not often that I have to count digits to see the size of a change!

This builds and tests cleanly on my box, and builds on the win-build machine.  I will stick to this branch for a while before merging it into "develop", because changes like this are tricky to undo.

The bottom line is to "Suggest" other packages, instead of using "ImportFrom".  So, the changes show up in DESCRIPTION, in various roxygen spots, and in the code itself ... which is why this is different enough that I will not merge for a while.

**Comment by j-harbin** [comment]
Dear reporter,

Do you think this issue (as defined by its title) has been addressed? If so, please close it. If not, please add a comment explaining what remains to be done. Thanks!

PS. This is a standardized reply.

**Comment by dankelley** [comment]
Jeeze, I can **never** remember that I was the one to make an issue!  Yup, it's done, so closing!
